memo.rb

```diff
@@ -0,0 +1,93 @@
+#!/usr/bin/env ruby
+# frozen_string_literal: true
+
+# myapp.rb
+
+require 'csv'
+require 'securerandom'
+require 'sinatra'
+require 'sinatra/reloader'
+
+MEMOS_CSV = 'memos.csv'
+
+helpers do
+  def h(text)
+    Rack::Utils.escape_html(text)
+  end
+end
+
+def read_memos
+  CSV.read(MEMOS_CSV)
+end
+
+def write_memo(memo)
+  CSV.open(MEMOS_CSV, 'a') { |memos| memos << memo }
+end
+
+def get_memo(id)
+  memo_values = []
+  CSV.foreach(MEMOS_CSV) do |memo|
+    memo_values = memo if memo[0] == id
+  end
+  memo_values
+end
+
+def replace_memos(edited_memos)
+  CSV.open(MEMOS_CSV, 'w') do |memos|
+    edited_memos.each do |edited_memo|
+      memos << edited_memo
+    end
+  end
+end
+
+get '/' do
+  @title = 'メモ一覧'
+  @memos = read_memos
+  erb :index
+end
+
+get '/memo/*' do |id|
+  @memo = {}
+  @memo[:id], @memo[:title], @memo[:text] = get_memo(id)
```

Comment:
```
  @memo[:id], @memo[:title], @memo[:text], @memo[:something] = get_memo(id)
```

項目が増えると結局、↑こんなふうに項目が増えていってしまうので、 get_memo 自身がハッシュを返すようにした方がいいですね。

```suggestion
  @memo = get_memo(id)
```